nbgrader/docs/source/configuration/jupyterhub_config.rst

```diff
@@ -195,3 +200,135 @@ if students may be enrolled in multiple courses, you need to provide them a
+    }
+
+    # instructor1 and instructor2 have access to different shared servers:
+    c.JupyterHub.load_groups = {
+        'formgrade-course101': [
+            'instructor1',
+            'grader-course101',
+        ],
+        'formgrade-course123': [
+            'instructor2',
+            'grader-course123'
+        ],
+        'nbgrader-course101': [],
+        'nbgrader-course123': []
+    }
+
+    # Start the notebook server as a service. The port can be whatever you want
+    # and the group has to match the name of the group defined above.
+    c.JupyterHub.services = [
+        {
+            'name': 'course101',
+            'url': 'http://127.0.0.1:9999',
+            'command': [
+                'jupyterhub-singleuser',
+                '--group=formgrade-course101',
+                '--debug',
+            ],
+            'user': 'grader-course101',
+            'cwd': '/home/grader-course101',
+            'api_token': ''  # include api token from admin user
+        },
+        {
+            'name': 'course123',
+            'url': 'http://127.0.0.1:9998',
+            'command': [
+                'jupyterhub-singleuser',
+                '--group=formgrade-course123',
+                '--debug',
+            ],
+            'user': 'grader-course123',
+            'cwd': '/home/grader-course123',
+            'api_token': ''  # include api token from admin user
+        },
+    ]
+
+
+Note: As you can see the ``nbgrader-{course_id}`` group is an empty list, adding students to the jupyterhub group is automatically done when the instructor adds them to the course database with the ``nbgrader db student add`` command or through the formgrader.
+
+
+There also needs to be a ``nbgrader_config.py`` file in the root of each grader
+account, which points to the directory where the class files are, e.g. in
+``/home/grader-course101/.jupyter/nbgrader_config.py`` would be:
+
+.. code:: python
+
+    c = get_config()
+    c.CourseDirectory.root = '/home/grader-course101/course101'
+
+and ``/home/grader-course123/.jupyter/nbgrader_config.py`` would be:
+
+.. code:: python
+
+    c = get_config()
+    c.CourseDirectory.root = '/home/grader-course123/course123'
+
+
+
+
+Custom Authentication
+^^^^^^^^^^^^^^^^^^
+
+To make your own custom authentication such as through an LTI you could start by making a method that inherits the Authenticator class, which is a plugin for different authentication methods.
+
+There are now three authentication classes:
+
+* ``NoAuthPlugin`` : The default old behaviour. Using this plugin will allow any user to any course if he does not have a course_id in his nbgrader_config. This is still the default behaviour so no need to specify it in ``etc/jupyter/nbgrader_config.py``
+
+* ``JupyterHubAuthPlugin`` : Uses the Jupyterhub groups part of the Jupyterhub API for authentication.
+
+* ``Authenticator``: Inherit this class to make your own Authenticator plugin, thought of as a way to enable LTI use cases.
```

Comment:
Actually this isn't quite right---to have your own plugin you have to inherit from `BaseAuthPlugin`. The `Authenticator` class is the object which is configurable to have different plugins.

Could you also add here some documentation about the methods that have to be overwritten? These are `get_student_courses`, `add_student_to_course`, and `remove_student_from_course`. You can do this automatically with sphinx directives, something like this:

```
.. currentmodule:: nbgrader.auth

.. autoclass:: BaseAuthPlugin

    .. automethod:: get_student_courses

    .. automethod:: add_student_to_course

    .. automethod:: remove_student_from_course
```

Reply:
I added a normal rst table for the methods that need to be overridden, I could not get the extensions in nbgrader/docs/source/conf.py to work with this. There was no currentmodule extensions found, I tried adding autosummary, it accepted that extension but I had some errors.